zerver/lib/saved_replies.py

```diff
@@ -0,0 +1,56 @@
+from typing import Any
+
+from django.utils.translation import gettext as _
+
+from zerver.lib.exceptions import ResourceNotFoundError
+from zerver.lib.markdown import markdown_convert
+from zerver.models import SavedReply, UserProfile
+from zerver.tornado.django_api import send_event_on_commit
+
+
+def do_create_saved_reply(
+    title: str,
+    content: str,
+    user_profile: UserProfile,
+) -> SavedReply:
+    title = title.strip()[: SavedReply.MAX_TITLE_LENGTH]
+    content = content.strip()
+
+    # Used in the UI in dropdown for showing the rendered version of the content in a single line.
+    rendered_content = markdown_convert(
+        content.replace("\n", " "), message_realm=user_profile.realm, no_previews=True
+    ).rendered_content
```

Comment:
Why are you replacing newlines with spaces here? I guess the motivation seems to be for fitting these into a limited area... Can you do some more extensive screenshots, maybe in the design thread for the feature, of various more complex syntax in a saved reply (bulleted lists / images / etc.). My guess is some of those will be corrupted and this is undesirable.

(We can manage space in other ways in the UI; simplest would be to just use CSS to cut off the region displayed)

Reply:
I'm also not 100% sure we want to do Markdown rendering in the server, vs. having the web app render these when loading the view, like we do with Drafts.